src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -396,9 +395,13 @@ void PhaseIdealLoop::get_assertion_predicates(Node* predicate, Unique_Node_List&
 IfProjNode* PhaseIdealLoop::clone_assertion_predicate_for_unswitched_loops(Node* iff, IfProjNode* predicate,
                                                                            Deoptimization::DeoptReason reason,
                                                                            ParsePredicateSuccessProj* parse_predicate_proj) {
-  Node* bol = create_bool_from_template_assertion_predicate(iff, nullptr, nullptr, parse_predicate_proj);
+  Node* opaque4_node = iff->in(1);
+  TemplateAssertionPredicateBool template_assertion_predicate_bool(opaque4_node->in(1));
+  BoolNode* bol = template_assertion_predicate_bool.clone(parse_predicate_proj, this);
+  opaque4_node = clone_and_register(opaque4_node, parse_predicate_proj);
```

Comment:
Why do you now clone the opaque node here, and not as part of `template_assertion_predicate_bool.clone`?